### ISSUE TITLE
selection is triggered even upon short right clicking

### ISSUE THREAD

**Issue by thaliel** [cpriest/SnapLinksPlus#432]
I have noticed this recently on my most current firefox: I right click something like an image or a link, and in addion to the context menu, the selection of snaplinks will also be triggered. This is getting rather annyoing when just doing a normal right click and not holding the right mouse button. I have changed the settings to have selection triggered by left-clicking for now, but this behaviour has never haoppened before and is therefore quite odd

**Comment by cpriest** [comment]
What FF Version & OS/Version are you using?

**Comment by thaliel** [comment]
> What FF Version & OS/Version are you using?

I am on FF version 102.0.1 on windows 10 64bit

**Comment by remi-garcia** [comment]
See https://github.com/cpriest/SnapLinksPlus/issues/400#issuecomment-1003716437

Maybe a recent update corrected the way context menu opens to match what it is suppose to do in settings.

In `about:config` could you try to toggle `ui.context_menus.after_mouseup` to `true` to see if that fix this issue?

**Comment by thaliel** [comment]
> See [#400 (comment)](https://github.com/cpriest/SnapLinksPlus/issues/400#issuecomment-1003716437)
> 
> Maybe a recent update corrected the way context menu opens to match what it is suppose to do in settings.
> 
> In `about:config` could you try to toggle `ui.context_menus.after_mouseup` to `true` to see if that fix this issue?

that seemes to have helped for a short time, but now te behaviour is the same again

**Comment by cpriest** [comment]
> that seemes to have helped for a short time, but now te behaviour is the same again

Strange that it would have changed its behavior for only a short time, did the `about:config` get changed back to its prior value?

**Comment by thaliel** [comment]
I changed it  back myself. I have now increased the amount of pixels that need to be dragged, which at least reduces how often this happens, but it is still quite annoying

**Comment by daf0x** [comment]
I've recently switched from Windows to Linux, and I think I have the same issue, `Ubuntu 23.10`, `Firefox 118.0.1 (64-bit)`. This never happened before on windows (and still doesn't on this PC, dual booting), so it seems to be related to Linux.

I noticed that this happens on many, but not all sites. For example, it happens 100% of the time on the google search results page, but doesn't seem to happen on a Bugzilla page.

I further noticed that on the google search page very rarely there will be a snaplinks outline drawn from the top-left corner of the screen to the point of the click for a fraction of a second, which then immediately disappears again, and the selection is active and starts from the point of right click (under the pop-up menu). I set the `Drag for N pixels` setting to a ridiculous value like > 1000 pixels, and then the selection does not start on the search results page any more. So it might be that somehow snaplinks thinks that there is an initial drag from the top-left of the window to the point of the click that isn't really there.

**Comment by cpriest** [comment]
Yeah this is an old thorny issue with how Linux/Firefox handles clicks differently on Linux (on the mousedown rather than the mouseup or visa versa)
